sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_client_connection.py

```diff
@@ -3272,11 +3271,12 @@ def _UpdateSessionIfRequired(
     def _get_partition_key_definition(self, collection_link: str) -> Optional[Dict[str, Any]]:
         partition_key_definition = None
         # If the document collection link is present in the cache, then use the cached partitionkey definition
-        if collection_link in self.partition_key_definition_cache:
-            partition_key_definition = self.partition_key_definition_cache.get(collection_link)
+        if collection_link in self.collection_properties_cache:
+            cached_collection: Dict[str, Any] = self.collection_properties_cache.get(collection_link)
+            partition_key_definition = cached_collection.get("partitionKey")
         # Else read the collection from backend and add it to the cache
         else:
             collection = self.ReadContainer(collection_link)
             partition_key_definition = collection.get("partitionKey")
-            self.partition_key_definition_cache[collection_link] = partition_key_definition
```

Comment:
Can you include somewhere on the description what the key, **collection_link**, and value, **collection**, expects to look like so I can assess the size?